### PR TITLE
Added mocks for Illager and Pillager

### PR DESCRIPTION
**Pull request by 4everTheOne**
# Description
Implemented mocks for [Illager](https://jd.papermc.io/paper/1.21.1/org/bukkit/entity/Illager.html) and [Pillager](https://jd.papermc.io/paper/1.21.1/org/bukkit/entity/Pillager.html).

A small refactor in InventoryMock was required, to support this change. In Paper, the Inventory does not have a restriction in sizes, only in CraftServer.
To resolve this, the validation was moved to ChestInventory.

# Checklist
The following items should be checked before the pull request can be merged.
- [X] Code follows existing style.
- [X] Unit tests added (if applicable).

# Relates
- https://github.com/MockBukkit/MockBukkit/issues/825

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/IllagerMock.java
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/IllagerMock.java:1]: Mostly a question, there's no different implementation for illagers?

File: src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java [hunk 1]
```diff
@@ -90,7 +91,7 @@
 import org.spigotmc.event.player.PlayerSpawnLocationEvent;
 
 import javax.imageio.ImageIO;
-import java.awt.Graphics2D;
+import java.awt.*;
```
Comment (Thorinwasher) [inline src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java]: I often do this as well, anyhow please avoid wildcard imports. You can change this in your IDE settings in such a way that it does not autoformat to this.

File: src/main/java/be/seeseemelk/mockbukkit/entity/PillagerMock.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * Mock implementation of a {@link Pillager}.
+ *
+ * @see IllagerMock
+ */
+public class PillagerMock extends IllagerMock implements Pillager, MockRangedEntity<PillagerMock>
+{
+
+	private final Inventory inventory = new InventoryMock(this, 5, InventoryType.CHEST);
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PillagerMock.java:25]: I don't know that much about inventories, but from looking into the paper source code I found that the main hand should have a crossbow item in it for pillagers.

**Review by Thorinwasher — CHANGES_REQUESTED** [review]
I found one small inconsistency between the paper implementation and this implementation (To be honest here, that one was kind of hard to find, but I'm always suspicious when the implementation is too similar to the super class)

For more info: I think `net.minecraft.world.entity.Entity#finalizeSpawn()` is always triggered when spawning an entity, this method essentially adds a crossbow to the main hand equipment slot of the entity. The thing is that these differences are only visible in mojang code.

File: src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java (hunk 1 above)
Comment (4everTheOne) [reply src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java]: Thanks for the heads up. For some reason in this project I still had the configuration to covert the "java.awt" to a wildcard import. I've removed this and updated the PR.

File: src/main/java/be/seeseemelk/mockbukkit/entity/IllagerMock.java
Comment (Thorinwasher) [reply src/main/java/be/seeseemelk/mockbukkit/entity/IllagerMock.java:1]: I could not find anything special in the mojang implementation, so I assume not

Comment (4everTheOne) [reply src/main/java/be/seeseemelk/mockbukkit/entity/IllagerMock.java:1]: In Paper/Bukkit implemention the Illager (CraftIllager) is working like an abstract class that will be extended by [Evoker](https://jd.papermc.io/paper/1.21.1/org/bukkit/entity/Evoker.html), [Illusioner](https://jd.papermc.io/paper/1.21.1/org/bukkit/entity/Illusioner.html), [Pillager](https://jd.papermc.io/paper/1.21.1/org/bukkit/entity/Pillager.html), [Spellcaster](https://jd.papermc.io/paper/1.21.1/org/bukkit/entity/Spellcaster.html), [Vindicator](https://jd.papermc.io/paper/1.21.1/org/bukkit/entity/Vindicator.html).

File: src/main/java/be/seeseemelk/mockbukkit/entity/PillagerMock.java (hunk 2 above)
Comment (4everTheOne) [reply src/main/java/be/seeseemelk/mockbukkit/entity/PillagerMock.java:25]: Good catch! I totally forgot about the Crossbow.
PR has been updated with it.

**Review by Thorinwasher — APPROVED** [review]
Looks good to me

**Pull request merged**